### PR TITLE
Changed diff to xmldiff

### PR DESCRIPTION
**Pull request by zonArt**
making it a little simplier to compare and removing the substitution as it is ok for xmldiff to deal with them correctly, adds a dependency on https://www.logilab.org/project/xmldiff

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -25,5 +25,6 @@
     { "name": "darin/zypprepo", "version_requirement": ">= 1.0.1 < 2.0.0" },
     { "name": "nanliu/staging", "version_requirement": ">= 1.0.0 < 2.0.0" },
     { "name": "camptocamp/archive", "version_requirement": ">= 0.8.0" }
+    { "name": "logilab/xmldiff", "version_requirement": ">=0.6.10" },
```
Comment (jhoblitt) [inline metadata.json]: There's no such module published on the forge.

Comment (zonArt) [inline metadata.json]: You're right, I didn't know if the dependencies were only for puppet modules or for every third library. To make this modification a python module is needed, it is called xmldiff (https://pypi.python.org/pypi/xmldiff). What is your suggestion to reflect this needs, modification of the readme ?

**Comment by jhoblitt** [comment]
@zonArt I'm not really keen on installing a package via pip since there may not be a working python env.  Does #418 cover your needs?

**Comment by zonArt** [comment]
@jhoblitt Ok I understand you're point and I agree that it probably isn't the best option. #418 looks good but as the new types are still in beta, I'm not really confident to use them now for our production instance, though I'll take a look at them. At least could I modify this PR to grant the ability to chose the difftool used ?

**Comment by zonArt** [comment]
#422 to replace this one

**Pull request closed without merging**